### ISSUE TITLE
Render a route inside an index route

### ISSUE THREAD

**Issue by startupthekid** [rackt/react-router#2856]
I have the following route configuration:

```
<Route path="/" component={App}>
    <IndexRoute component={Promote} />
    <Route path="earn" component={Earn}/>
    <Route path="vision" component={Vision}/>
    <Route path="contact" component={Contact}/>
    <Route path="demo" component={Demo}/>
    <Route path="tos" component={Legal}/>
    <Route path="privacy" component={Legal}/>
    <Route path="dmca" component={Legal}/>
  </Route>
```

It's pretty simple, when the page loads at `/`, the `App` component is loaded with a default page of `Promote`. What I want to accomplish is having a route inside the `IndexRoute` so that if a user visits `/child-route`, it'll navigate to that component inside the `IndexRoute` component. However I still want that component to always be visible inside it's parent (the `IndexRoute` `Promote` component). I really just need a way to always have it render but also have it be linkable as a sub-component.

I tried configuring my routes like this:

```
<Route path="/" component={App}>
    <IndexRoute component={Promote}>
      <Route path="signup" component={SignupBar} />
    </IndexRoute>
    <Route path="earn" component={Earn}/>
    <Route path="vision" component={Vision}/>
    <Route path="contact" component={Contact}/>
    <Route path="demo" component={Demo}/>
    <Route path="tos" component={Legal}/>
    <Route path="privacy" component={Legal}/>
    <Route path="dmca" component={Legal}/>
  </Route>
```
but that didn't work because from how I understand it that would only render a signup component when `/signup` is active and that's not what I want.

Basically, I want the `SignupBar` component to be part of the `render` method of `Promote` but be able to be navigated to with whatever url I choose.

Would the best way to use an `onEnter` hook on the `IndexRoute` and then manually scroll down to where I need to be if the url is `/signup`?

**Comment by timdorr** [comment]
Thanks for your question!

We want to make sure that the GitHub issue tracker remains the best place to track bug reports and feature requests that affect the development of React Router.

Questions like yours deserve a purpose-built Q&A forum. Could you post this question to Stack Overflow with the tag #react-router? https://stackoverflow.com/questions/ask?tags=react-router.

We also have an active and helpful React Router community on Reactiflux, which is a great place to get fast help with React Router and with the rest of the React ecosystem. You can join at https://discord.gg/0ZcbPKXt5bYaNQ46.

**Comment by petr001** [comment]
dealing with the same problem, did you find any solution?

**Comment by Gregoirevda** [comment]
UP
